src/main/java/br/unicamp/cst/motivational/MotivationalSubsystemViewer.java

```diff
@@ -49,24 +48,32 @@
     private DefaultTreeModel dtAppraisalCodelets;
     private DefaultTreeModel dtMoodCodelets;
     private DefaultTreeModel dtGoalCodelets;
+    private Mind m;
     /**
      * Creates new form MotivationalSubsystemViewer
      */
     public MotivationalSubsystemViewer(long refreshTime, Mind mind) {
         initComponents();
         setRefreshTime(refreshTime);
-        initMotivationalSubsystemViewer(mind.getMotivationalSubsystemModule().getMotivationalCodelets(), 
-                                        mind.getMotivationalSubsystemModule().getEmotionalCodelets(), 
-                                        mind.getMotivationalSubsystemModule().getGoalCodelets(), 
-                                        mind.getMotivationalSubsystemModule().getAppraisalCodelets(), 
-                                        mind.getMotivationalSubsystemModule().getMoodCodelets());
+//        initMotivationalSubsystemViewer(mind.getMotivationalSubsystemModule().getMotivationalCodelets(), 
+//                                        mind.getMotivationalSubsystemModule().getEmotionalCodelets(), 
+//                                        mind.getMotivationalSubsystemModule().getGoalCodelets(), 
+//                                        mind.getMotivationalSubsystemModule().getAppraisalCodelets(), 
+//                                        mind.getMotivationalSubsystemModule().getMoodCodelets());
+        initMotivationalSubsystemViewer(mind.getGroupList("Motivational"), 
+                                        mind.getGroupList("Emotional"),
+                                        mind.getGroupList("Goal"),
+                                        mind.getGroupList("Appraisal"),
+                                        mind.getGroupList("Moode")
+                                        );
+        m = mind;
     }
     
-    private void initMotivationalSubsystemViewer(List<MotivationalCodelet> motivationalCodelets,
-                                                List<EmotionalCodelet> emotionalCodelets,
-                                                List<GoalCodelet> goalCodelets,
-                                                List<AppraisalCodelet> appraisalCodelets,
-                                                List<MoodCodelet> moodCodelets) {
+    private void initMotivationalSubsystemViewer(List<Codelet> motivationalCodelets,
```

Comment:
This code smells - the fact that this method has 5 arguments (best practices point to maximum 4 allowed) is the clue. We have 5 different lists of the same type `Codelet`. What happens if the architecture evolves to have yet another different kind of `Codelet` list? Probably we would have to add yet another argument to this method. This deisgn needs to be refactored. If they are all `Codelets`, why not a single `List`, or even a `Map<String, Codelet>`, if you want to force the developer to somehow differentiate from them?